From 44f21f022edbdf0032b9f6d2e197d7e042c14d04 Mon Sep 17 00:00:00 2001
From: Axel Lin <axel.lin@gmail.com>
Date: Sat, 26 Nov 2011 12:55:43 +0800
Subject: mmc: convert drivers/mmc/host/* to use module_platform_driver()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch converts the drivers in drivers/mmc/host/* to use the
module_platform_driver() macro which makes the code smaller and a bit
simpler.

Signed-off-by: Axel Lin <axel.lin@gmail.com>
Acked-by: "Michał Mirosław" <mirq-linux@rere.qmqm.pl>
Acked-by: David Brown <davidb@codeaurora.org>
Acked-by: Viresh Kumar <viresh.kumar@st.com>
Acked-by: Guennadi Liakhovetski <g.liakhovetski@gmx.de>
Acked-by: Sascha Hauer <s.hauer@pengutronix.de>
Acked-by: Wolfram Sang <w.sang@pengutronix.de>
Acked-by: Anton Vorontsov <cbouatmailru@gmail.com>
Signed-off-by: Chris Ball <cjb@laptop.org>
(cherry picked from commit d1f81a64a4250bdd776978be06ae2b8e13ec7471)

Conflicts:

	drivers/mmc/host/bfin_sdh.c
	drivers/mmc/host/cb710-mmc.c
	drivers/mmc/host/jz4740_mmc.c
	drivers/mmc/host/msm_sdcc.c
	drivers/mmc/host/mxcmmc.c
	drivers/mmc/host/mxs-mmc.c
	drivers/mmc/host/pxamci.c
	drivers/mmc/host/s3cmci.c
	drivers/mmc/host/sdhci-cns3xxx.c
	drivers/mmc/host/sdhci-dove.c
	drivers/mmc/host/sdhci-esdhc-imx.c
	drivers/mmc/host/sdhci-of-esdhc.c
	drivers/mmc/host/sdhci-of-hlwd.c
	drivers/mmc/host/sdhci-pxav2.c
	drivers/mmc/host/sdhci-pxav3.c
	drivers/mmc/host/sdhci-s3c.c
	drivers/mmc/host/sdhci-spear.c
	drivers/mmc/host/sdhci-tegra.c
	drivers/mmc/host/sh_mobile_sdhi.c
	drivers/mmc/host/tmio_mmc.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/mmc/host/sh_mmcif.c | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/drivers/mmc/host/sh_mmcif.c b/drivers/mmc/host/sh_mmcif.c
index 86edbad..fd0fd11 100644
--- a/drivers/mmc/host/sh_mmcif.c
+++ b/drivers/mmc/host/sh_mmcif.c
@@ -1207,19 +1207,7 @@ static struct platform_driver sh_mmcif_driver = {
 	},
 };
 
-static int __init sh_mmcif_init(void)
-{
-	return platform_driver_register(&sh_mmcif_driver);
-}
-
-static void __exit sh_mmcif_exit(void)
-{
-	platform_driver_unregister(&sh_mmcif_driver);
-}
-
-module_init(sh_mmcif_init);
-module_exit(sh_mmcif_exit);
-
+module_platform_driver(sh_mmcif_driver);
 
 MODULE_DESCRIPTION("SuperH on-chip MMC/eMMC interface driver");
 MODULE_LICENSE("GPL");
-- 
1.7.10.2.565.gbd578b5

